### PR TITLE
Cleanup GitLabWebHook

### PR DESCRIPTION
**Pull request by coder-hugo**
This PR cleans the code of the GitLabWebHook by splitting it into multiple classes.
The PR is based on #198.

### CODE REVIEW

File: src/main/java/com/dabsquared/gitlabjenkins/webhook/build/PushBuildAction.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                        buildOpenMergeRequests(trigger, pushRequest.getProject_id(), pushRequest.getRef());
+                    }
+                }
+            }
+        });
+    }
+
+    protected void buildOpenMergeRequests(final GitLabPushTrigger trigger, final Integer projectId, String projectRef) {
+        try {
+            GitLabConnectionProperty property = project.getProperty(GitLabConnectionProperty.class);
```
Comment (daddykotex) [inline src/main/java/com/dabsquared/gitlabjenkins/webhook/build/PushBuildAction.java:67]: Ok, you got it fixed here!

File: src/main/java/com/dabsquared/gitlabjenkins/webhook/build/PushBuildAction.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                        ACL.impersonate(ACL.SYSTEM, new Runnable() {
+                            public void run() {
+                                trigger.onPost(createMergeRequest(projectId, mergeRequest, lastCommit));
+                            }
+                        });
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.warning("failed to communicate with gitlab server to determine is this is an update for a merge request: " + e.getMessage());
```
Comment (jtheoof) [inline src/main/java/com/dabsquared/gitlabjenkins/webhook/build/PushBuildAction.java]: minor: _to determine is_ should be: _to determine if_

**Comment by coder-hugo** [comment]
@slide, @markus-mnm: @omehegan told me that you are more experienced in Java/Jenkins plugin development than he. Could you please have a look to my changes before I merge them back? As you can see in the description of the PR this is based on another PR (which is also based on another PR :wink:) So you can decide whether you want to review my whole changes or to review smaller changes by reviewing the PRs in that order: #192, #198, this one.
If you like you can already have a look to #203 which will be one of my next cleanups.

Thanks,
coder-hugo

**Comment by coder-hugo** [comment]
Close this as there is no possibility to change the target branch. New PR #231

**Comment by markus-mnm** [comment]
I'll have a look at this this weekend, but I might not provide as much value as one might hope.
( posted this to the wrong PR, but started looking at the right PR #231)

**Pull request closed without merging**